libr/anal/p/anal_mips_gnu.c

```diff
@@ -277,46 +1446,138 @@ static int mips_op(RAnal *anal, RAnalOp *op, ut64 addr, const ut8 *b, int len, R
 			op->val = imm;
 			break;
 		case 9: // addiu
+			insn.id = MIPS_INS_ADDIU;
+			op->type = R_ANAL_OP_TYPE_ADD;
 			op->dst = r_anal_value_new ();
 			op->dst->reg = r_reg_get (anal->reg, mips_reg_decode(rt), R_REG_TYPE_GPR);
 			// TODO: currently there is no way for the macro to get access to this register
 			op->src[0] = r_anal_value_new ();
 			op->src[0]->reg = r_reg_get (anal->reg, mips_reg_decode(rs), R_REG_TYPE_GPR);
 			op->val = imm; // Beware: this one is signed... use `?vi $v`
+			if (rs == 0) {
+				insn.id = MIPS_INS_LI;
+				sprintf (insn.i_reg.imm, "0x%"PFMT32x, imm);
+			}
 			break;
 		case 8: // addi
-		case 10: // stli
-		case 11: // stliu
+			insn.id = MIPS_INS_ADDI;
+			op->type = R_ANAL_OP_TYPE_ADD;
+			break;
+		case 10: // slti
+			insn.id = MIPS_INS_SLTI;
+			break;
+		case 11: // sltiu
+			insn.id = MIPS_INS_SLTIU;
+			break;
 		case 12: // andi
+			insn.id = MIPS_INS_ANDI;
+			op->type = R_ANAL_OP_TYPE_AND;
+			break;
 		case 13: // ori
+			insn.id = MIPS_INS_ORI;
+			op->type = R_ANAL_OP_TYPE_OR;
+			break;
 		case 14: // xori
+			insn.id = MIPS_INS_XORI;
+			op->type = R_ANAL_OP_TYPE_XOR;
+			break;
+		case 24: // daddi
+			insn.id = MIPS_INS_DADDI;
+			op->type = R_ANAL_OP_TYPE_ADD;
+			break;
+		case 25: // daddiu
+			insn.id = MIPS_INS_DADDIU;
+			op->type = R_ANAL_OP_TYPE_ADD;
+			if (rs == 0) {
+				insn.id = MIPS_INS_LDI;
+				sprintf (insn.i_reg.imm, "0x%"PFMT64x, imm);
+			}
+			break;
 		case 32: // lb
+			op->refptr =  1;
+			insn.id = MIPS_INS_LB;
 		case 33: // lh
+			if (!op->refptr) {
+				op->refptr =  2;
+				insn.id = MIPS_INS_LB;
+			}
 		case 35: // lw
+			if (!op->refptr) {
+				op->refptr =  4;
+				insn.id = MIPS_INS_LW;
+			}
+		case 55: // ld
+			if (!op->refptr) {
+				op->refptr =  8;
+				insn.id = MIPS_INS_LD;
+			}
+			
+			if (rs == 28){
+				op->ptr = anal->gp + imm;
+			}else{
+				op->ptr = imm;
+			}
+			if (rt == 25){
+				t9_pre = op->ptr;
+			}
+			op->type = R_ANAL_OP_TYPE_LOAD;
+			break;
 		case 36: // lbu
+			insn.id = MIPS_INS_LBU;
+			op->type = R_ANAL_OP_TYPE_LOAD;
+			break;
 		case 37: // lhu
+			insn.id = MIPS_INS_LHU;
+			op->type = R_ANAL_OP_TYPE_LOAD;
+			break;
 		case 40: // sb
+			insn.id = MIPS_INS_SB;
+			op->type = R_ANAL_OP_TYPE_STORE;
+			break;
 		case 41: // sh
+			insn.id = MIPS_INS_SH;
+			op->type = R_ANAL_OP_TYPE_STORE;
+			break;
 		case 43: // sw
+			insn.id = MIPS_INS_SW;
+			op->type = R_ANAL_OP_TYPE_STORE;
+			break;
+		case 63: //sd
+			insn.id = MIPS_INS_SD;
+			op->type = R_ANAL_OP_TYPE_STORE;
+			break;
 		case 49: // lwc1
 		case 57: // swc1
 			break;
 		case 29: // jalx
+			insn.id = MIPS_INS_JALX;
 			op->type = R_ANAL_OP_TYPE_CALL;
 			op->jump = addr + 4*((buf[3] | buf[2]<<8 | buf[1]<<16));
 			op->fail = addr + 8;
 			op->delay = 1;
+			sprintf (insn.i_reg.jump, "0x%lx", op->jump);
+
 			break;
 		}
 		//family = 'I';
 	}
 
+	if (mask & R_ANAL_OP_MASK_ESIL) {
+		if (analop_esil (anal, op, addr, &insn) != 0) {
```

Comment:
!=0 is implicit, you can remove this check. but more importantly. change the function to return bool instead of int